WordPress/src/main/java/org/wordpress/android/ui/comments/CommentAdapter.java

```diff
@@ -409,36 +470,36 @@ protected void onCancelled() {
         }
         @Override
         protected Boolean doInBackground(Void... params) {
-            if (mStatusFilter == null){
-                tmpComments = CommentTable.getCommentsForBlogWithFilter(mLocalBlogId, CommentStatus.UNKNOWN);
+            List<CommentModel> comments;
+            if (mStatusFilter == null || mStatusFilter == CommentStatus.ALL) {
+                // The "all" filter actually means "approved" + "unapproved" (but not "spam", "trash" or "deleted")
+                comments = mCommentStore.getCommentsForSite(mSite, CommentStatus.APPROVED, CommentStatus.UNAPPROVED);
             } else {
-                tmpComments = CommentTable.getCommentsForBlogWithFilter(mLocalBlogId, mStatusFilter);
+                comments = mCommentStore.getCommentsForSite(mSite, mStatusFilter);
             }
 
-            if (mComments.isSameList(tmpComments)) {
-                return false;
-            }
+            tmpComments = new CommentList();
+            tmpComments.addAll(comments);
 
-            // pre-calc transient values so they're cached prior to display
-            for (Comment comment: tmpComments) {
-                comment.getDatePublished();
-                comment.getUnescapedPostTitle();
-                comment.getAvatarForDisplay(mAvatarSz);
-                comment.getFormattedTitle();
-
-                String content = StringUtils.notNullStr(comment.getCommentText());
-                //to load images embedded within comments, pass an ImageGetter to WPHtml.fromHtml()
-                Spanned spanned = WPHtml.fromHtml(content, null, null, mContext, null, 0);
-                comment.setUnescapedCommentWithDrawables(spanned);
-            }
-
-            return true;
+            return !mComments.isSameList(tmpComments);
         }
         @Override
         protected void onPostExecute(Boolean result) {
             if (result) {
                 mComments.clear();
                 mComments.addAll(tmpComments);
+                // Sort by date
```

Comment:
Here we're sorting the comments in descending order, but we're receiving them from the `CommentsModelTable` in [ascending order](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/4375dd239b95ceb48c20723635d3ab3c0931b804/fluxc/src/main/java/org/wordpress/android/fluxc/persistence/CommentSqlUtils.java#L115).

I see that we may have assembled `mComments` from multiple calls, so we always need to perform a sort here, but what do you think of changing `CommentSqlUtils` to return descending order by default? It should be more efficient in the case where `mComments` only contains a single call's worth of comments.